tests/tools.vitruv.applications.pcmjava.tests/src/tools/vitruv/applications/pcmjava/tests/pcm2java/Pcm2JavaTransformationTest.xtend

```diff
@@ -0,0 +1,192 @@
+package tools.vitruv.applications.pcmjava.tests.pcm2java
+
+import java.nio.file.Path
+import java.util.Collection
+import java.util.Comparator
+import java.util.HashSet
+import java.util.Set
+import java.util.stream.IntStream
+import org.eclipse.emf.common.util.ECollections
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.emf.ecore.util.EcoreUtil
+import org.eclipse.xtend.lib.annotations.Accessors
+import org.emftext.language.java.commons.NamedElement
+import org.emftext.language.java.containers.CompilationUnit
+import org.emftext.language.java.containers.Package
+import org.junit.jupiter.api.BeforeEach
+import org.palladiosimulator.pcm.repository.Repository
+import org.palladiosimulator.pcm.repository.RepositoryFactory
+import org.palladiosimulator.pcm.system.System
+import org.palladiosimulator.pcm.system.SystemFactory
+import tools.vitruv.applications.pcmjava.pcm2java.Pcm2JavaChangePropagationSpecification
+import tools.vitruv.applications.util.temporary.java.JavaSetup
+import tools.vitruv.change.propagation.ChangePropagationMode
+import tools.vitruv.framework.views.View
+import tools.vitruv.testutils.ViewBasedVitruvApplicationTest
+import tools.vitruv.testutils.printing.DefaultModelPrinter
+import tools.vitruv.testutils.printing.DefaultPrintIdProvider
+import tools.vitruv.testutils.printing.DefaultPrintTarget
+
+import static org.junit.jupiter.api.Assertions.*
+import static tools.vitruv.applications.pcmjava.tests.pcm2java.JavaQueryUtil.*
+import static tools.vitruv.applications.pcmjava.tests.pcm2java.TransformationDirectionConfiguration.configureUnidirectionalExecution
+
+class Pcm2JavaTransformationTest extends ViewBasedVitruvApplicationTest {
+	protected var extension Pcm2JavaViewFactory viewFactory
+	
+	// === setup ===
+	
+	@Accessors(PROTECTED_GETTER)
+	static val PCM_MODEL_FILE_EXTENSION = "pcm"
+	@Accessors(PROTECTED_GETTER)
+	static val PCM_SYSTEM_FILE_EXTENSION = "system"
+	@Accessors(PROTECTED_GETTER)
+	static val PCM_MODEL_NAME = "model"
+	@Accessors(PROTECTED_GETTER)
+	static val JAVA_MODEL_FILE_EXTENSION = "java"
+	@Accessors(PROTECTED_GETTER)
+	static val JAVA_MODEL_NAME = "code"
+	@Accessors(PROTECTED_GETTER)
+	static val MODEL_FOLDER_NAME = "model"
+	
```

Comment:
```suggestion
	@BeforeAll
	def static void setupJavaFactories() {
		JavaSetup.prepareFactories()
	}
```

This will fix the problem with the unresolved changes. After adding it, one test still fails as the expected and actual compilation units do not match. Maybe you can have a detailed look on this first. If you are not able to find the problem, just ping me again.